packages/react-native-views/src/components/Thumbnail/index.js

```diff
@@ -0,0 +1,146 @@
+  const { colors } = theme;
+  const { id } = metadata.sight;
+
+  const [complianceModalIsOpen, openComplianceModal, closeComplianceModal] = useToggle();
+  const complianceIssues = useMemo(
+    () => Object.keys(complianceStatus).filter(
+      (key) => !!complianceStatus[key],
+    ), [complianceStatus],
+  );
+  const isNotCompliant = complianceIssues?.length;
+
+  console.log({ complianceStatus, image, isUploading, uploadStatus });
+
+  const isFailed = uploadStatus === 'rejected';
+  const isPending = uploadStatus === 'pending';
+  return (
+    <>
+      <ComplianceModal
+        visible={complianceModalIsOpen}
+        complianceIssues={complianceIssues}
+        onDismiss={closeComplianceModal}
+      />
+      <View style={[styles.sightCard, { borderColor: colorsVariant(colors)[uploadStatus] }]}>
+        <View style={styles.reloadButtonLayout}>
+          {/* compliance button */}
+          {isNotCompliant ? (
+            <IconButton
+              icon="information-outline"
+              size={24}
+              onPress={openComplianceModal}
+              style={styles.overlayButton}
+              color={colors.info}
+            />
+          ) : null}
+
+          {/* reload button */}
+          {!isPending && isFailed ? (
+            <IconButton
+              icon="reload"
+              size={24}
+              onPress={() => null}
+              style={styles.overlayButton}
+              color={colors.error}
+            />
+          ) : null}
+        </View>
+
+        {/* sight mask */}
+        {uploadStatus !== 'pending' ? (
+          <View style={{ transform: [{ scale: 0.19 }], zIndex: 2, height: 400 }}>
+            <SightMask id={id.charAt(0).toUpperCase() + id.slice(1)} height="400" width="500" color={colorsVariant(colors)[uploadStatus]} />
+          </View>
+        ) : null}
+
+        {/* we can try implementing the new Img conponent here for a smooth image rendering */}
+        {image
+          ? <Image source={{ uri: image }} style={styles.picture} />
+          : null}
+
+        {/* loading */}
+        {uploadStatus === 'pending' ? <ActivityIndicator style={styles.loading} color={uploadStatus === 'rejected' ? colors.error : colors.primary} /> : null}
+      </View>
+    </>
+  );
+}
+
+Thumbnail.propTypes = {
+  complianceStatus: PropTypes.shape({
+    hasBlur: PropTypes.bool,
+    isOverExposed: PropTypes.bool,
+    isUnderExposed: PropTypes.bool,
+  }),
+  image: PropTypes.shape({
+    height: PropTypes.number,
+    source: PropTypes.string,
+    width: PropTypes.number,
+  }).isRequired,
+  isUploading: PropTypes.bool,
+  metadata: PropTypes.shape({ sight: propTypes.sight }).isRequired,
+  uploadStatus: PropTypes.oneOf(['pending', 'fulfilled', 'rejected']),
+};
+Thumbnail.defaultProps = {
+  complianceStatus: { hasBlur: false, isOverExposed: false, isUnderExposed: false },
+  isUploading: PropTypes.bool,
+  uploadStatus: 'pending',
+};
+export default withTheme(Thumbnail);
```

Comment:
This is clean work, but why the use of `withTheme` instead of `useTheme` ? Is it because the hook has some issues working in a package ? (I had issues like that)

Reply:
I remember I had an issue with it before, and didn't spend more time on it, I just used withTheme so i can move on, let me try it with the hook and see.